types/json-query/index.d.ts

```diff
@@ -0,0 +1,58 @@
+// Type definitions for json-query 2.2
+// Project: http://github.com/mmckegg/json-query#readme
+// Definitions by: Matt Traynham <https://github.com/mtraynham>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = JsonQuery;
+
+declare function JsonQuery(
+    selector: JsonQuery.Selector | JsonQuery.SelectorWithQueryParams,
+    options: JsonQuery.Options
+): JsonQuery.Result;
+
+declare namespace JsonQuery {
+    type Selector = string;
+
+    type QueryParam = any;
+    // No way to support [Selector, ...QueryParam[]]?
```

Comment:
Maybe `[Selector] & { [x: number]: QueryParam }`

Reply:
I thought I tried that but came up empty handed.  Should that work?